File: go/creationaltype/builder/director.go
package builder

type Director struct {
	builder Builder
}

func NewDirector(builder Builder) *Director {
	return &Director{
		builder: builder,
	}
}

// Construct 控制执行流程
func (d *Director) Construct() Product {
	d.builder.initProduct()
	d.builder.builderPart1()
	d.builder.builderPart2()
	return d.builder.getResult()
}

File: java/creationalType/builder/src/Director.java
public class Director {
    // Director控制builder的执行顺序
    // builder负责具体的执行逻辑

    Builder builder;

    Director(Builder builder) {
        this.builder = builder;
    }

    // 指挥者的指挥构建顺序
    public Product construct() {
        this.builder.init();
        this.builder.buildPart1();
        this.builder.buildPart2();
        return this.builder.getResult();
    }

}
